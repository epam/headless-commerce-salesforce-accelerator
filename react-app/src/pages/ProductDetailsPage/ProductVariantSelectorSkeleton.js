/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import ContentLoader from "react-content-loader";
import PropTypes from "prop-types";

const VariantSelectorSkeleton = ({ isBundle = false }) => (
  <ContentLoader viewBox="0 0 900 900" width="100%" speed={2}>
    <circle cx="15" cy="15" r="15" />
    <circle cx="55" cy="15" r="15" />
    <circle cx="95" cy="15" r="15" />
    <circle cx="135" cy="15" r="15" />

    <rect x="0" y="50" rx="0" ry="0" width="70%" height="32" />
    <rect x="0" y="95" rx="0" ry="0" width="20%" height="20" />

    <rect x="0" y="155" rx="0" ry="0" width="10%" height="50" />

    {isBundle ? (
      <>
        <rect x="0" y="235" rx="0" ry="0" width="10%" height="20" />
        <rect x="0" y="265" rx="0" ry="0" width="100%" height="10" />
        <rect x="0" y="300" rx="0" ry="0" width="20%" height="150" />

        <rect x="200" y="300" rx="0" ry="0" width="25%" height="20" />
        <rect x="200" y="340" rx="0" ry="0" width="15%" height="20" />
        <rect x="200" y="380" rx="0" ry="0" width="30%" height="20" />
        <rect x="200" y="420" rx="0" ry="0" width="25%" height="20" />

        <rect x="0" y="475" rx="0" ry="0" width="100%" height="10" />

        <rect x="0" y="510" rx="0" ry="0" width="20%" height="150" />

        <rect x="200" y="510" rx="0" ry="0" width="25%" height="20" />
        <rect x="200" y="550" rx="0" ry="0" width="15%" height="20" />
        <rect x="200" y="590" rx="0" ry="0" width="30%" height="20" />
        <rect x="200" y="630" rx="0" ry="0" width="25%" height="20" />

        <rect x="0" y="680" rx="0" ry="0" width="100%" height="10" />

        <rect x="0" y="730" rx="0" ry="0" width="10%" height="20" />
        <rect x="0" y="760" rx="0" ry="0" width="20%" height="50" />

        <rect x="0" y="840" rx="0" ry="0" width="30%" height="80" />
        <rect x="300" y="840" rx="0" ry="0" width="10%" height="80" />
      </>
    ) : (
      <>
        <rect x="0" y="235" rx="0" ry="0" width="15%" height="20" />
        <rect x="0" y="275" rx="0" ry="0" width="70" height="50" />

        <rect x="80" y="275" rx="0" ry="0" width="70" height="50" />
        <rect x="0" y="360" rx="0" ry="0" width="5%" height="20" />

        <rect x="0" y="390" rx="0" ry="0" width="30%" height="60" />
        <rect x="0" y="490" rx="0" ry="0" width="10%" height="20" />

        <rect x="0" y="560" rx="0" ry="0" width="10%" height="20" />
        <rect x="0" y="600" rx="0" ry="0" width="20%" height="60" />

        <rect x="0" y="690" rx="0" ry="0" width="30%" height="80" />
        <rect x="300" y="690" rx="0" ry="0" width="10%" height="80" />
      </>
    )}
  </ContentLoader>
);

VariantSelectorSkeleton.propTypes = {
  isBundle: PropTypes.bool,
};

export default VariantSelectorSkeleton;
