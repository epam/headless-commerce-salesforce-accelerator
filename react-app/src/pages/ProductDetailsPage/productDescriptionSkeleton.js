/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import ContentLoader from "react-content-loader";

const DescriptionSkeleton = () => (
  <ContentLoader viewBox="0 0 1200 100" width="100%" height={100} speed={2}>
    <rect x="0" y="0" rx="0" ry="0" width="20%" height={20} />
    <rect x="0" y="30" rx="0" ry="0" width="100%" />
  </ContentLoader>
);

export default DescriptionSkeleton;
