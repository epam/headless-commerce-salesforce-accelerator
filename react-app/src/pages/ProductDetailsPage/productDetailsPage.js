/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import PropTypes from "prop-types";

import Grid from "@material-ui/core/Grid";

import productActions from "store/actions/product/actions";
import miniCartActions from "store/actions/cart/actions";
import {
  selectIsProductLoading,
  selectProductId,
  selectIsAvailableToOrder,
  selectProductQuantities,
  selectProductName,
  selectProductOption,
} from "store/selectors/product/selectors";

import Carousel from "components/ProductDetails/Carousel";
import SliderSkeleton from "components/core/SliderSkeleton";
import VariantSelectors from "components/ProductDetails/VariantSelectors";
import ScrollToTopController from "components/core/ScrollToTopController";
import Description from "components/ProductDetails/Description";
import Details from "components/ProductDetails/Details";
import Attributes from "components/ProductDetails/Attributes";
import ProductDetailsContainer from "./productDetailsContainer";
import VariantSelectorSkeleton from "./ProductVariantSelectorSkeleton";
import DescriptionSkeleton from "./productDescriptionSkeleton";
import Button from "../../components/core/Button";
import BagIcon from "../../components/core/ShoppingBagIcon";

const ProductDetailsPage = ({ match }) => {
  const dispatch = useDispatch();
  const isLoading = useSelector(selectIsProductLoading);
  const productId = useSelector(selectProductId);
  const isAvailableToOrder = useSelector(selectIsAvailableToOrder);
  const quantities = useSelector(selectProductQuantities);
  const productName = useSelector(selectProductName);
  const productOption = useSelector(selectProductOption);

  useEffect(() => {
    dispatch(productActions.getProductById(match.params.pid));
  }, [dispatch, match.params.pid]);

  const onAddToCart = () => {
    dispatch(
      miniCartActions.addProductToCart(
        productId,
        quantities.find((value) => value.selected).value,
        productName,
        productOption
      )
    );
  };

  return (
    <>
      <ScrollToTopController />
      {productId && (
        <ProductDetailsContainer>
          <Grid container spacing={3}>
            {isLoading ? (
              <>
                <Grid item xs={6}>
                  <SliderSkeleton />
                </Grid>
                <Grid item xs={6}>
                  <VariantSelectorSkeleton isBundle />
                </Grid>
                <Grid item xs={12}>
                  <DescriptionSkeleton />
                </Grid>
              </>
            ) : (
              <>
                <Grid item xs={6}>
                  <Carousel />
                </Grid>
                <Grid item xs={6}>
                  <VariantSelectors />
                  <Button
                    disabled={!isAvailableToOrder}
                    onClick={onAddToCart}
                    icon={<BagIcon />}
                    margin="24px 0 0 0"
                  >
                    Add to Cart
                  </Button>
                </Grid>
                <Grid item xs={12}>
                  <Description />
                  <Details />
                  <Attributes />
                </Grid>
              </>
            )}
          </Grid>
        </ProductDetailsContainer>
      )}
    </>
  );
};

ProductDetailsPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      pid: PropTypes.string.isRequired,
    }),
  }).isRequired,
};

export default ProductDetailsPage;
