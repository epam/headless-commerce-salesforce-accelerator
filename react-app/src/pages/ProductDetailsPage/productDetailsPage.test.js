/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import PropTypes from "prop-types";

import ProductDetailsPage from "./productDetailsPage";
import ProductDetailsContainer from "./productDetailsContainer";
import DescriptionSkeleton from "./productDescriptionSkeleton";
import VariantSelectorSkeleton from "./ProductVariantSelectorSkeleton";
import getMainTheme from "../../theme";

const ReduxProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

ReduxProvider.propTypes = {
  children: PropTypes.element,
  reduxStore: PropTypes.shape({}),
};

const theme = getMainTheme();

describe("ProductDetailsPage", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <ReduxProvider>
        <ProductDetailsPage />
      </ReduxProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("ProductDetailsContainer", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<ProductDetailsContainer theme={theme} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe("DescriptionSkeleton", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <ReduxProvider>
        <DescriptionSkeleton />
      </ReduxProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("VariantSelectorSkeleton", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <ReduxProvider>
        <VariantSelectorSkeleton isBundle />
      </ReduxProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
