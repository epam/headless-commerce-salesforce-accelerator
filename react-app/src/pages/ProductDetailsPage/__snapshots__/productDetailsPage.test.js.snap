// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionSkeleton should match snapshot 1`] = `
<Provider>
  <DescriptionSkeleton />
</Provider>
`;

exports[`ProductDetailsContainer should match snapshot 1`] = `
<div
  className="sc-fznyAO JWwbI"
/>
`;

exports[`ProductDetailsPage should match snapshot 1`] = `
<Provider>
  <ProductDetailsPage />
</Provider>
`;

exports[`VariantSelectorSkeleton should match snapshot 1`] = `
<Provider>
  <VariantSelectorSkeleton
    isBundle={true}
  />
</Provider>
`;
